### PR TITLE
Add support for WLCG Bearer tokens

### PR DESCRIPTION
**Pull request by oshadura**
Discovery schema: https://github.com/WLCG-AuthZ-WG/bearer-token-discovery/blob/master/specification.md
(adopted based on the implementation in htcondor/htcondor.git by @bbockelm)

Fixes: https://github.com/root-project/root/issues/6802
(based on https://github.com/root-project/root/pull/6818)

### CODE REVIEW

File: net/davix/inc/TDavixFile.h [hunk 1]
```diff
@@ -107,6 +115,11 @@ class TDavixFile : public TFile {
     /// Enable or disable certificate authority check
     void setCACheck(Bool_t check);
 
+    // Determine the value of the current token from the process's environment.
+    // Follows the WLCG Bearer Token Discovery schema.
+    // On error or no token discovered, returns the empty string.
+    std::string discoverToken();
```
Comment (pcanal) [inline net/davix/inc/TDavixFile.h:121]:
```suggestion
    std::string DiscoverToken();
```

File: net/davix/src/TDavixFile.cxx [hunk 2]
```diff
@@ -142,6 +150,118 @@ static void ConfigureDavixLogLevel()
...
+
+// Token discovery process is based on WLCG specification:
+// https://github.com/WLCG-AuthZ-WG/bearer-token-discovery/blob/master/specification.md
+std::string discoverToken()
+{
+	const char *bearer_token = getenv("BEARER_TOKEN");
+	std::string token;
+	if (bearer_token && *bearer_token)
+	{
+		if (!normalizeToken(bearer_token, token)) {return "";}
```
Comment (pcanal) [inline net/davix/src/TDavixFile.cxx:233]: This file might be contains tab character, if it is the case, can you replace them with (3) spaces?

Comment (oshadura) [reply net/davix/src/TDavixFile.cxx]: @pcanal ops and thanks!

File: net/davix/src/TDavixFile.cxx [hunk 3]
```diff
@@ -411,12 +527,21 @@ void TDavixFileInternal::parseConfig()
...
    if (gDebug > 0)
       Info("parseConfig", "Setting CAcheck to %s", ((ca_check_local) ? ("true") : ("false")));
 
+   // WLCG Bearer tokens check
+   std::string prefix = "Bearer ";
+   auto token = DiscoverToken();
+   if (token.c_str()) {
+      // header: "Authorization: Bearer mytoken"
+      davixParam->addHeader("Authorization", prefix + token);
+    }
```
Comment (pcanal) [inline net/davix/src/TDavixFile.cxx:543]:
```suggestion
      davixParam->addHeader("Authorization", prefix + token);
   }
```

**Comment by pcanal** [comment]
Can you also update the release and include there a summary of the specification?

**Comment by pcanal** [comment]
@phsft-bot build

**Review by bbockelm — APPROVED** [review]
@oshadura - thank you so much for taking this on!  It's much appreciated.

**Comment by pcanal** [comment]
> Can you also update the release and include there a summary of the specification?

@oshadura ping?

File: README/ReleaseNotes/v624/index.md [hunk 4]
```diff
@@ -121,6 +121,11 @@ For details, see the RooFit tutorial [rf612_recoverFromInvalidParameters.C](http
 
 ## Networking Libraries
 
+- Enabled WLCG Bearer Tokens support in RDavix. Bearer tokens are part of WLCG capability-based infrastructure with capability-based scheme uses an infrastructure that describes what the bearer is allowed to do as opposed to who that bearer is. Token discovery procedure are developed according WLCG Bearer Token Discovery specification document (https://github.com/WLCG-AuthZ-WG/bearer-token-discovery/blob/master/specification.md). Short overview:
+   1. If the BEARER_TOKEN environment variable is set, then the value is taken to be the token contents.
+   2. If the BEARER_TOKEN_FILE environment variable is set, then its value is interpreted as a filename. The contents of the specified file are taken to be the token contents.
+   3. If the XDG_RUNTIME_DIR environment variable is set*, then take the token from the contents of $XDG_RUNTIME_DIR/bt_u$ID**.
```
Comment (pcanal) [inline README/ReleaseNotes/v624/index.md:127]: is the placement of the starts in `set*` vs `..ID**` intentional (one star followed by 2 stars)?

File: README/ReleaseNotes/v624/index.md [hunk 5]
```diff
@@ -121,6 +121,11 @@ For details, see the RooFit tutorial [rf612_recoverFromInvalidParameters.C](http
 
 ## Networking Libraries
 
+- Enabled WLCG Bearer Tokens support in RDavix. Bearer tokens are part of WLCG capability-based infrastructure with capability-based scheme uses an infrastructure that describes what the bearer is allowed to do as opposed to who that bearer is. Token discovery procedure are developed according WLCG Bearer Token Discovery specification document (https://github.com/WLCG-AuthZ-WG/bearer-token-discovery/blob/master/specification.md). Short overview:
```
Comment (pcanal) [inline README/ReleaseNotes/v624/index.md:124]:
```suggestion
- Enabled WLCG Bearer Tokens support in RDavix. Bearer tokens are part of WLCG capability-based infrastructure with capability-based scheme which uses an infrastructure that describes what the bearer is allowed to do as opposed to who that bearer is. Token discovery procedure are developed according WLCG Bearer Token Discovery specification document (https://github.com/WLCG-AuthZ-WG/bearer-token-discovery/blob/master/specification.md). Short overview:
```

**Comment by bbockelm** [comment]
@oshadura - are the build failures above related to the PR?

**Comment by oshadura** [comment]
@bbockelm no they are not related! I was waiting for the last word from @pcanal to merge this PR.

**Comment by oshadura** [comment]
@pcanal ping.

**Review by pcanal — APPROVED** [review]

**Pull request merged**